File: clas12/ctof/hit.pl
use strict;
use warnings;

our %configuration;


my @allConfs = ("original", "cad");

sub define_hit
{
	foreach my $conf ( @allConfs )
	{
		$configuration{"variation"} = $conf ;

		# uploading the hit definition
		my %hit = init_hit();
		$hit{"name"}            = "ctof";
		$hit{"description"}     = "ctof hit definitions";
		$hit{"identifiers"}     = "paddle";
		$hit{"signalThreshold"} = "0.5*MeV";
		$hit{"timeWindow"}      = "5*ns";
		$hit{"prodThreshold"}   = "2*mm";
		$hit{"maxStep"}         = "1*cm";
		$hit{"delay"}           = "50*ns";
		$hit{"riseTime"}        = "5*ns";
		$hit{"fallTime"}        = "8*ns";
		$hit{"mvToMeV"}         = 100;
		$hit{"pedestal"}        = -20;
		print_hit(\%configuration, \%hit);
	}
}

File: basic_examples/3_Working_with_geometry/ex3_2_parameters_and_variations/hit.py
from gemc_api_hit import *

def define_hit(configuration):
	# uploading the hit definition
	hit = MyHit(name="ctof", description="ctof hit definitions", identifiers="paddle", SignalThreshold="0.5*MeV",
		TimeWindow="5*ns", ProdThreshold="1*mm", MaxStep="1*cm", delay="50*ns",
		riseTime="5*ns", fallTime="8*ns", mvToMeV=100, pedestal=-20)
	
	print_hit(configuration, hit)
